### ISSUE TITLE
OS.has_touchscreen_ui_hint reports false on Steam Deck

### ISSUE THREAD

**Issue by mnn** [godotengine/godot#68518]
### Godot version

3.5.1.stable.official

### System information

Steam Deck

### Issue description

Docs:

> Returns `true` if the device has a touchscreen or emulates one.

`OS.has_touchscreen_ui_hint` reports `false` on Steam Deck, despite the device having a touchscreen.

### Steps to reproduce

Call `OS.has_touchscreen_ui_hint` on Steam Deck.

### Minimal reproduction project

```gdscript
func _ready() -> void:
	print("Touchscreen: %s" % [OS.has_touchscreen_ui_hint()])
```

**Comment by Akylzhan** [comment]
Are you using Linux on Steam Deck? From what I see Godot just checks if touch emulation from mouse is turned on:

https://github.com/godotengine/godot/blob/28afd32d29df78283d0433680812c70edfb14c65/core/os/os.cpp#L443-L446

On Windows it is properly implemented though.
I might be wrong since I am not an expert in Godot codebase

**Comment by mnn** [comment]
Yes, Linux - SteamOS I think (the official preinstalled thing from Valve based on Arch). I was trying it in "game mode" (which players are most likely be using), not the desktop mode.

I think it does some emulation too - but it's probably not mouse emulating touch input (maybe pads are emulating mouse?).

**Comment by Akylzhan** [comment]
@mnn I meant, Godot has an option to emulate touch from mouse in Project Settings, try turning it on

**Comment by Calinou** [comment]
To my knowledge, touchscreen recognition isn't implemented on Linux yet. It's not easy to get working, unlike most other platforms.

**Comment by adamscott** [comment]
For more information, see [Arch's wiki page](https://wiki.archlinux.org/title/touchscreen) on the subject. Does not seem simple.
